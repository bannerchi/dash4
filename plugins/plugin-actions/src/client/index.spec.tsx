// /// <reference types="@types/jest" />
import { render } from '@testing-library/react';
import React from 'react';
import { PluginActions } from './index';

jest.mock('@dash4/log/build/browser', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	error: jest.fn(),
	warn: jest.fn(),
	success: jest.fn(),
}));

const configLink = {
	actions: [
		{
			id: '1',
			type: 'link' as 'link',
			href: 'http://localhost:6006',
			title: 'Storybook',
			image: 'https://storybook.js.org/images/logos/icon-storybook.png',
		},
		{
			id: '2',
			type: 'link' as 'link',
			href: 'http://localhost:6008',
			title: 'Development environment',
			icon: 'compare',
		},
		{
			id: '3',
			type: 'link' as 'link',
			href: 'http://localhost:6009',
			title: 'Staging environment',
			icon: 'devices',
		},
	],
};
const configTeaser = {
	actions: [
		{
			id: 'teaser-1',
			type: 'teaser' as 'teaser',
			href: 'http://localhost:6006',
			title: 'Storybook',
			subtitle:
				'Build bulletproof UI components faster. Storybook is an open source tool for developing UI components in isolation for React, Vue, and Angular. It makes building stunning UIs organized and efficient.',
			image: 'https://storybook.js.org/images/logos/icon-storybook.png',
			link: configLink.actions[0],
		},
		{
			id: 'teaser-2',
			type: 'teaser' as 'teaser',
			href: 'http://localhost:6006',
			title: 'Storybook',
			subtitle:
				'Build bulletproof UI components faster. Storybook is an open source tool for developing UI components in isolation for React, Vue, and Angular. It makes building stunning UIs organized and efficient.',
			image: 'https://storybook.js.org/images/logos/icon-storybook.png',
			link: configLink.actions,
		},
		{
			id: 'teaser-3',
			type: 'teaser' as 'teaser',
			href: 'http://localhost:6006',
			title: 'Storybook',
			subtitle:
				'Build bulletproof UI components faster. Storybook is an open source tool for developing UI components in isolation for React, Vue, and Angular. It makes building stunning UIs organized and efficient.',
			icon: 'compare',
			link: configLink.actions[0],
		},
	],
};

const props = {
	id: '1',
	name: 'PluginActions',
	lowerCaseName: 'plugin-actions',
	dark: false,
};

describe('PluginTerminal', () => {
	test('exist', () => {
		expect(typeof PluginActions).toBe('object');
	});
	test('links', () => {
		const { container } = render(
			<PluginActions
				{...{
					...props,
					clientConfig: {
						title: 'Links',
						...configLink,
					},
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('links in dark mode', () => {
		const { container } = render(
			<PluginActions
				{...{
					...props,
					dark: true,
					clientConfig: {
						title: 'Links',
						...configLink,
					},
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('teaser with one link', () => {
		const { container } = render(
			<PluginActions
				{...{
					...props,
					dark: true,
					clientConfig: {
						actions: [configTeaser.actions[0]],
					},
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('teaser with multiple links', () => {
		const { container } = render(
			<PluginActions
				{...{
					...props,
					clientConfig: {
						actions: [configTeaser.actions[1]],
					},
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('teaser with icon', () => {
		const { container } = render(
			<PluginActions
				{...{
					...props,
					clientConfig: {
						actions: [configTeaser.actions[2]],
					},
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
