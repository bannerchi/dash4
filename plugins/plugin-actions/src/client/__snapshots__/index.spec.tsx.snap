// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginTerminal links 1`] = `
.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  color: #000;
  white-space: pre;
  background: #fff;
  border: 1px solid #fff;
  box-shadow: rgba(0,0,0,0.16) 0px 2px 4px;
  border-radius: 6px;
  height: 100%;
}

.emotion-12:focus {
  border: 1px solid var(--primary);
  outline: none;
}

.emotion-1 {
  position: relative;
  padding: 15px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  padding: 15px 15px 0;
}

.emotion-0 {
  width: 100%;
  white-space: normal;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2;
  margin-bottom: 0;
  overflow: hidden;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.emotion-11 {
  padding: 5px;
  white-space: normal;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 15px;
}

.emotion-10 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--dark);
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
}

.emotion-3:hover,
.emotion-3:focus,
.emotion-3:active {
  color: var(--dark);
  -webkit-text-decoration: none;
  text-decoration: none;
  background: rgba(108,117,125,0.1);
}

.emotion-2 {
  margin: 3px 10px 4px 0;
  max-width: 18px;
  max-height: 18px;
}

.emotion-5 {
  position: relative;
  display: block;
  width: 18px;
  height: 18px;
  margin: 3px 10px 4px 0;
}

.emotion-4 {
  font-size: 18px;
}

<div
  class="emotion-12"
  tabindex="1"
>
  <div
    class="emotion-1"
  >
    <p
      class="emotion-0"
    >
      Links
    </p>
  </div>
  <div
    class="emotion-11"
  >
    <ul
      class="emotion-10"
    >
      <li>
        <a
          class="emotion-3 btn btn-link"
          href="http://localhost:6006"
          rel="noreferrer noopener"
          target="_blank"
        >
          <img
            alt="Storybook"
            class="emotion-2"
            src="https://storybook.js.org/images/logos/icon-storybook.png"
          />
          Storybook
        </a>
      </li>
      <li>
        <a
          class="emotion-3 btn btn-link"
          href="http://localhost:6008"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            class="emotion-5"
          >
            <i
              class="material-icons emotion-4"
            >
              compare
            </i>
          </span>
          Development environment
        </a>
      </li>
      <li>
        <a
          class="emotion-3 btn btn-link"
          href="http://localhost:6009"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            class="emotion-5"
          >
            <i
              class="material-icons emotion-4"
            >
              devices
            </i>
          </span>
          Staging environment
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PluginTerminal links in dark mode 1`] = `
.emotion-1 {
  position: relative;
  padding: 15px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  padding: 15px 15px 0;
}

.emotion-0 {
  width: 100%;
  white-space: normal;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2;
  margin-bottom: 0;
  overflow: hidden;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.emotion-11 {
  padding: 5px;
  white-space: normal;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 15px;
}

.emotion-10 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-2 {
  margin: 3px 10px 4px 0;
  max-width: 18px;
  max-height: 18px;
}

.emotion-5 {
  position: relative;
  display: block;
  width: 18px;
  height: 18px;
  margin: 3px 10px 4px 0;
}

.emotion-4 {
  font-size: 18px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  color: #fff;
  white-space: pre;
  background: #000;
  border: 1px solid rgb(51,51,51);
  box-shadow: rgba(0,0,0,0.2) 0px 20px 50px 0px;
  border-radius: 5px;
  height: 100%;
}

.emotion-12:focus {
  border: 1px solid var(--primary);
  outline: none;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--white);
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
}

.emotion-3:hover,
.emotion-3:focus,
.emotion-3:active {
  color: var(--white);
  -webkit-text-decoration: none;
  text-decoration: none;
  background: var(--gray-dark);
}

<div
  class="emotion-12"
  tabindex="1"
>
  <div
    class="emotion-1"
  >
    <p
      class="emotion-0"
    >
      Links
    </p>
  </div>
  <div
    class="emotion-11"
  >
    <ul
      class="emotion-10"
    >
      <li>
        <a
          class="emotion-3 btn btn-link"
          href="http://localhost:6006"
          rel="noreferrer noopener"
          target="_blank"
        >
          <img
            alt="Storybook"
            class="emotion-2"
            src="https://storybook.js.org/images/logos/icon-storybook.png"
          />
          Storybook
        </a>
      </li>
      <li>
        <a
          class="emotion-3 btn btn-link"
          href="http://localhost:6008"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            class="emotion-5"
          >
            <i
              class="material-icons emotion-4"
            >
              compare
            </i>
          </span>
          Development environment
        </a>
      </li>
      <li>
        <a
          class="emotion-3 btn btn-link"
          href="http://localhost:6009"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            class="emotion-5"
          >
            <i
              class="material-icons emotion-4"
            >
              devices
            </i>
          </span>
          Staging environment
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PluginTerminal teaser with icon 1`] = `
.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  color: #000;
  white-space: pre;
  background: #fff;
  border: 1px solid #fff;
  box-shadow: rgba(0,0,0,0.16) 0px 2px 4px;
  border-radius: 6px;
  height: 100%;
}

.emotion-10:focus {
  border: 1px solid var(--primary);
  outline: none;
}

.emotion-9 {
  padding: 5px;
  white-space: normal;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 15px;
}

.emotion-8 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-7 {
  background: transparent;
  border: 0;
}

.emotion-5 {
  padding: 12px 0;
}

.emotion-3 {
  padding: 0 12px;
}

.emotion-4 {
  padding: 0 12px;
}

.emotion-6 {
  position: absolute;
  display: block;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.emotion-2 {
  padding: 12px 0 20px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  position: relative;
  display: block;
  width: 24px;
  height: 24px;
}

.emotion-0 {
  font-size: 24px;
}

<div
  class="emotion-10"
  tabindex="1"
>
  <div
    class="emotion-9"
  >
    <ul
      class="emotion-8"
    >
      <li>
        <div
          class="emotion-7 card"
        >
          <div
            class="emotion-2"
          >
            <span
              class=" emotion-1"
            >
              <i
                class="material-icons emotion-0"
              >
                compare
              </i>
            </span>
          </div>
          <div
            class="emotion-5 card-body"
          >
            <div
              class="emotion-3 card-title h5"
            >
              Storybook
            </div>
            <p
              class="emotion-4 card-text"
            >
              Build bulletproof UI components faster. Storybook is an open source tool for developing UI components in isolation for React, Vue, and Angular. It makes building stunning UIs organized and efficient.
            </p>
          </div>
          <a
            class="emotion-6"
            href="http://localhost:6006"
            rel="noreferrer noopener"
            target="_blank"
          />
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PluginTerminal teaser with multiple links 1`] = `
.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  color: #000;
  white-space: pre;
  background: #fff;
  border: 1px solid #fff;
  box-shadow: rgba(0,0,0,0.16) 0px 2px 4px;
  border-radius: 6px;
  height: 100%;
}

.emotion-16:focus {
  border: 1px solid var(--primary);
  outline: none;
}

.emotion-15 {
  padding: 5px;
  white-space: normal;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 15px;
}

.emotion-14 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--dark);
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
}

.emotion-5:hover,
.emotion-5:focus,
.emotion-5:active {
  color: var(--dark);
  -webkit-text-decoration: none;
  text-decoration: none;
  background: rgba(108,117,125,0.1);
}

.emotion-4 {
  margin: 3px 10px 4px 0;
  max-width: 18px;
  max-height: 18px;
}

.emotion-7 {
  position: relative;
  display: block;
  width: 18px;
  height: 18px;
  margin: 3px 10px 4px 0;
}

.emotion-6 {
  font-size: 18px;
}

.emotion-13 {
  background: transparent;
  border: 0;
}

.emotion-1 {
  padding: 12px 0 20px 0;
  text-align: center;
}

.emotion-0 {
  max-width: 100px;
  max-height: 100px;
}

.emotion-12 {
  padding: 12px 0;
}

.emotion-2 {
  padding: 0 12px;
}

.emotion-3 {
  padding: 0 12px;
}

<div
  class="emotion-16"
  tabindex="1"
>
  <div
    class="emotion-15"
  >
    <ul
      class="emotion-14"
    >
      <li>
        <div
          class="emotion-13 card"
        >
          <div
            class="emotion-1"
          >
            <img
              alt="Storybook"
              class="card-img-top emotion-0"
              src="https://storybook.js.org/images/logos/icon-storybook.png"
            />
          </div>
          <div
            class="emotion-12 card-body"
          >
            <div
              class="emotion-2 card-title h5"
            >
              Storybook
            </div>
            <p
              class="emotion-3 card-text"
            >
              Build bulletproof UI components faster. Storybook is an open source tool for developing UI components in isolation for React, Vue, and Angular. It makes building stunning UIs organized and efficient.
            </p>
            <a
              class="emotion-5 btn btn-link"
              href="http://localhost:6006"
              rel="noreferrer noopener"
              target="_blank"
            >
              <img
                alt="Storybook"
                class="emotion-4"
                src="https://storybook.js.org/images/logos/icon-storybook.png"
              />
              Storybook
            </a>
            <a
              class="emotion-5 btn btn-link"
              href="http://localhost:6008"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span
                class="emotion-7"
              >
                <i
                  class="material-icons emotion-6"
                >
                  compare
                </i>
              </span>
              Development environment
            </a>
            <a
              class="emotion-5 btn btn-link"
              href="http://localhost:6009"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span
                class="emotion-7"
              >
                <i
                  class="material-icons emotion-6"
                >
                  devices
                </i>
              </span>
              Staging environment
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PluginTerminal teaser with one link 1`] = `
.emotion-8 {
  padding: 5px;
  white-space: normal;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 15px;
}

.emotion-7 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  color: #fff;
  white-space: pre;
  background: #000;
  border: 1px solid rgb(51,51,51);
  box-shadow: rgba(0,0,0,0.2) 0px 20px 50px 0px;
  border-radius: 5px;
  height: 100%;
}

.emotion-9:focus {
  border: 1px solid var(--primary);
  outline: none;
}

.emotion-6 {
  background: transparent;
  border: 0;
}

.emotion-1 {
  padding: 12px 0 20px 0;
  text-align: center;
}

.emotion-0 {
  max-width: 100px;
  max-height: 100px;
}

.emotion-4 {
  padding: 12px 0;
}

.emotion-2 {
  padding: 0 12px;
}

.emotion-3 {
  padding: 0 12px;
}

.emotion-5 {
  position: absolute;
  display: block;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

<div
  class="emotion-9"
  tabindex="1"
>
  <div
    class="emotion-8"
  >
    <ul
      class="emotion-7"
    >
      <li>
        <div
          class="emotion-6 card"
        >
          <div
            class="emotion-1"
          >
            <img
              alt="Storybook"
              class="card-img-top emotion-0"
              src="https://storybook.js.org/images/logos/icon-storybook.png"
            />
          </div>
          <div
            class="emotion-4 card-body"
          >
            <div
              class="emotion-2 card-title h5"
            >
              Storybook
            </div>
            <p
              class="emotion-3 card-text"
            >
              Build bulletproof UI components faster. Storybook is an open source tool for developing UI components in isolation for React, Vue, and Angular. It makes building stunning UIs organized and efficient.
            </p>
          </div>
          <a
            class="emotion-5"
            href="http://localhost:6006"
            rel="noreferrer noopener"
            target="_blank"
          />
        </div>
      </li>
    </ul>
  </div>
</div>
`;
